### DISCUSSION TITLE
Scrolling

### DISCUSSION THREAD

**Discussion by lucasrodwell** [hakimel/reveal.js#3448] [General]
I am using the Obsdian advanced slides plugin which uses reveal.js. I often want to interactively add to presentations with my audience, which means that I cannot be sure not to overflow on a slide, and need to be able to scroll. Sometimes I have longer texts that I don't want to or can't split up, such as embeds, where it would be very helpful to be able to scroll. With Obsidian I am allowed to add a css "snippet", but I'm just a user, not a dev. Any suggestions?

**Comment by t-fritsch** [comment]
Hi,
Since this is not an issue with reveal and more an "help wanted" situation, I transfer this to the discussions 🙏

**Comment by t-fritsch** [comment]
I don't use Obsidian and thus don't know the options you have in it, and what the "css snippet" allows to customize, but in css when you want to limit the height of an element and add a scrollbar to it, you can do something like : 
```css
mySelector {
    max-height: 520px;
    overflow-y: auto;
}
```
what you need now is to find what `mySelector` should be to match the content created by Obsidian 😅 
you can try to right click on your content and then 'Inspect' it to find what css classes it has.

**Reply by lucasrodwell** [reply]
Thanks! Obsidian is a personal knowledge management system that uses Markdown (highly recommend – there are currently over 1000 plugins, of which [advanced slides](https://github.com/MSzturc/obsidian-advanced-slides) using reveal.js is one of them. Unfortunately, your solution only half works, because the text gets cut off at the top, like [this](https://stackoverflow.com/questions/49278725/centered-elements-inside-a-flex-container-are-growing-and-overflowing-beyond-top) I imagine. I can scroll down to the end, but not up to the top, Perhaps it would be easier to change something in the reveal.js code just to put scrollbars on the whole page or something like that ? The plugin downloads the latest version of reveal.js when it installs, so I guess it pretty much uses reveal.js out of the box. Any suggestions welcome !

**Reply by t-fritsch** [reply]
Thank you for the infos on Obsidian !
Regarding scrollbars I doubt this would be integrated in reveal (this is only my point of view, I don't decide anything here, only trying to provide some help) since imho the whole point of presentations is to serve as visual support for a speech, and thus get very low quantity of text, only the most important ideas. 

If one absolutely needs a lot of text, and then scrollbars, it is already possible to apply some css to any element in Reveal, and the css I posted yesterday should do the trick if applied to the element that contains the text (div, p). The behavior you described (text truncated at the top) is not normal, could you post a link to a codesandbox/jsfiddle/codepen that show the problem so that I can help you to find the source of the problem ?

**Comment by lucasrodwell** [comment]
I understand completely, it's just good to have the freedom to read a
longer text together or write down audience contributions without moving on
to another slide. I'm sending a few screenshots so you know what I'm
talking about. Sorry, I'm not a web programmer (I only know R), so I don't
know how to create a sandbox, but the link to the plugin is here:
https://github.com/MSzturc/obsidian-advanced-slides, and obsidian is here :
https://github.com/obsidianmd. Thanks for any tips !
